matplotlibrc.template

```diff
@@ -1,45 +1,25 @@
-#### MATPLOTLIBRC FORMAT
-
-## This is a sample matplotlib configuration file - you can find a copy
-## of it on your system in site-packages/matplotlib/mpl-data/matplotlibrc
-## (which related to your Python installation location).
-##
-## If you edit it there, please note that it will be overwritten in your
-## next install.  If you want to keep a permanent local copy that will not
-## be overwritten, place it in one of the following locations:
-## unix/linux:
-##     $HOME/.config/matplotlib/matplotlibrc OR
-##     $XDG_CONFIG_HOME/matplotlib/matplotlibrc (if $XDG_CONFIG_HOME is set)
-## other platforms:
-##     $HOME/.matplotlib/matplotlibrc
+## ############################################################################
+## # MATPLOTLIBRC CONFIGURATION FILE                                          #
+## ############################################################################
 ##
 ## See https://matplotlib.org/users/customizing.html#the-matplotlibrc-file
-## for more details on the paths which are checked for the configuration file.
-##
-## This file is best viewed in a editor which supports python mode syntax
-## highlighting.  Blank lines, or lines starting with a comment symbol, are
-## ignored, as are trailing comments.  Other lines must have the format:
-##     key : val  ## optional comment
+## for more details on path resolution and formats of this file.
 ##
-## Formatting and style conventions for this file:
-##     - at least one whitesapce AROUND `:` to seperate key and val
-##         * prefer one whitesapce except for block colon alignment
-##     - at least two whitesapces BEFORE `##` to seperate the key-val pair and
-##       the trailing comments
-##         * prefer two whitesapces except for block `##` alignment
-##     - at least one whitesapce AFTER `##` to seperate the `##` marker and
-##       the comment text
-##         * prefer one whitesapce except for indentation (listing, etc.), in
+## While modifying this file, one could follow (not required though) these
```

Comment:
Followup to https://github.com/matplotlib/matplotlib/pull/14920#discussion_r308602704, but could be done in a separate PR as well:
I would suggest modifying the style as follows:
- Remove the spaces *before* the colon (one can still use multiple spaces after the colon -- but at least one) for alignment purposes
- Use a single # for trailing comments, and have at least two spaces before them and one after them.

With these changes, the whole style suggestions then boil down to: "use normal PEP8-like style (as enforced in the rest of the codebase), except that all non-empty lines (whether including an rc entry or just comments must start with an *additional* #".

Reply:
As to the actual edit here: I think "Style conventions: ..." is good enough -- no need to write two lines for something that we're basically not going to strictly enforce.

Reply:
Good ideas! Since modifying the style would involves lots of lines change, I think it's better done in a separate PR. I can try and see the effects latter. Thanks!

Reply:
Just as a remark: Our current matplotlibrc file is a vaild YAML document.

It may be simpler for the user if we could state that matplotlibrc is a YAML file. For that, we would need to use an actual yaml parser instead of our custom parser in `matplotlib._rc_params_in_file`. On the plus side there are no definition and parsing ambiguities for the format. On the down side, we would need to add a dependency to PyYAML.

Note: This interferes with the floating idea of a python-based config file.